File: Article.cs

class Article
{
    int Reference;
    string Designation;
    float Prix;

    public Article()
    {
        this.Reference = Reference;
        this.Designation = Designation;
        this.Prix = Prix;
    }

    public Article(int Reference, string Designation, float Prix)
    {
        this.Reference = Reference;
        this.Designation = Designation;
        this.Prix = Prix;
    }

    public void Saisir()
    {
        Console.WriteLine("Donner la reference d'un article:");
        Reference = int.Parse(Console.ReadLine());
        Console.WriteLine("Donner la Designation:");
        Designation = Console.ReadLine();
        Console.WriteLine("Donner le prix:");
        Prix = float.Parse(Console.ReadLine());

    }
    
    public void Afficher()
    {
        Console.WriteLine($" reference={Reference} --designation={Designation} --prix={Prix}");
    }
}

File: Article.java
public class Article{
    static int ID=0;
    String Nom;
    String Reference;
    double Prix;


    public Article(String Nom,double Prix){
        this.Nom=Nom;
        this.Prix =Prix;
    }


    public  int getID(){
        
        return ID++;
    }

    public String getReference(){
        return Reference;
    }

    public String getNom(){
        return Nom;
    }

    public double getPrix(){
        return Prix;
    }


    
    public void setReference(String Reference){
        this.Reference=Reference;
    }

    public void setNom(String Nom){
        this.Nom=Nom;
    }

    public void setPrix(double Prix){
        this.Prix=Prix;
    }

    public String modifier(String Nom,int ID){
        StringBuilder Name=new StringBuilder();

        for(int i=0;i<Nom.length();i++){
             if(Nom.charAt(i)!=' ' &&i%5==0){
                Name.append(Nom.charAt(i));
             }

        }
        Name.append(ID);

        return Name.toString();

        

    }

    public String  modifier(String Nom,double Prix){
           return modifier(Nom, ID);
    }

    public String toString(){
        Reference=modifier(Nom, Prix);
        return "id= <"+ID+">, nom= <"+Nom+">, reference= <"+Reference+">, prix= <"+Prix+">";
    }




    public static void main(String[] args){
        Article A1=new Article("manette",1500);
        
        
        System.out.println(A1.toString());
    }
}